sql/core/src/main/scala/org/apache/spark/sql/execution/dynamicpruning/PartitionPruning.scala

```diff
@@ -229,6 +229,24 @@ object PartitionPruning extends Rule[LogicalPlan] with PredicateHelper {
     !plan.isStreaming && hasSelectivePredicate(plan)
   }
 
+  /**
+   * Check the partition has static filter. So we can skip add dynamic partition pruning
+   * if there already exists.
+   */
+  private def hasNoStaticPartitionEqualityCondition(e: Expression, p: LogicalPlan): Boolean = {
```

Comment:
Nit: use `hasStaticPartitionEqualityCondition` and reverses return values. It might be more readable to use `has` than `not had`. When calling this function, it will be more natural to write `if(hasStaticPartitionEqualityCondition)` and `if(!hasStaticPartitionEqualityCondition)`.

Reply:
thank you @amaliujia